modules/account-lib/src/coin/dot/transactionBuilder.ts

```diff
@@ -1,57 +1,480 @@
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  version(transactionVersion: number): this {
+    this._transactionVersion = transactionVersion;
+    return this;
+  }
+
+  /**
+   *
+   * The spec name for the registry.
+   *
+   * @param {PolkadotSpecNameType} specName
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  specName(specName: PolkadotSpecNameType): this {
+    this._specName = specName;
+    return this;
+  }
+
+  /**
+   *
+   * The genesis hash of the chain.
+   *
+   * @param {string} genesisHash
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  genesisHash(genesisHash: string): this {
+    this._genesisHash = genesisHash;
+    return this;
+  }
+
+  /**
+   *
+   * The raw metadata of the chain in string format.
+   *
+   * @param {string} metadataRpc
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  metadataRpc(metadataRpc: string): this {
+    this._metadataRpc = metadataRpc;
+    return this;
+  }
+
+  /**
+   *
+   * The specVersion for transaction format.
+   *
+   * @param {number} specVersion
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  specVersion(specVersion: number): this {
+    this._specVersion = specVersion;
+    return this;
+  }
+
+  /**
+   *
+   * The chainName for transaction format.
+   *
+   * @param {number} chainName
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  chainName(chainName: string): this {
+    this._chainName = chainName;
+    return this;
+  }
+
+  private method(method: TxMethod): this {
+    this._method = method;
+    return this;
+  }
+
+  /**
+   * Set the network based on the configuration in the statics module
+   */
+  private staticsConfig(): void {
+    const networkConfig = this._coinConfig.network as DotNetwork;
+    this.specName(networkConfig.specName);
+    this.genesisHash(networkConfig.genesisHash);
+    this.specVersion(networkConfig.specVersion);
+    this.chainName(networkConfig.chainName);
+  }
+
+  /**
+   * Sets the
+   * specName,
+   * genesisHash - genesisHash of the chain,
+   * metadataRpc - The SCALE-encoded metadata for the runtime when submitted,
+   * specVersion - The current spec version for the runtime,
+   * chainName - chainName,
+   * registry,
+   * of the westened
+   *
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  westend(): this {
```

Comment:
We shouldn't have these methods. We should get the network parameters from statics. Statics should pass a configuration with westend, testnet, or mainnet parameters